pkg/object/mem.go

```diff
@@ -129,16 +129,40 @@ func (m *memStore) Delete(key string) error {
 }
 
 func (m *memStore) List(prefix, marker, delimiter string, limit int64) ([]Object, error) {
-	if delimiter != "" {
-		return nil, notSupportedDelimiter
-	}
 	m.Lock()
 	defer m.Unlock()
 
 	objs := make([]Object, 0)
+	commonPrefixsMap := make(map[string]bool, 0)
 	for k := range m.objects {
 		if strings.HasPrefix(k, prefix) && k > marker {
 			o := m.objects[k]
+			if delimiter != "" {
+				remainString := strings.TrimPrefix(k, prefix)
+				pos := strings.Index(remainString, delimiter)
+				if pos != -1 && pos != len(remainString) {
+					commonPrefix := remainString[0:pos + 1]
+					if _, ok := commonPrefixsMap[commonPrefix]; ok {
+						continue
+					}
+					f := &file{
+						obj{
+							prefix + commonPrefix,
+							0,
+							time.Unix(0, 0),
+							strings.HasSuffix(commonPrefix, "/"),
+						},
+						"",
+						"",
+						0,
+						false,
+					}
+					objs = append(objs, f)
+					commonPrefixsMap[commonPrefix] = true
```

Comment:
This is not necessary.

Reply:
`prefix/a/a1/`
`prefix/a/`

The above two objects cause `a/` to be counted repeatedly, so I use commonPrefixMaps to repeat it here

Reply:
Just determine if the key exists

Reply:
ok